tests/main/nvidia-files/task.yaml

```diff
@@ -0,0 +1,18 @@
+summary: all files matching nvidia globs can be dlopened
+details: |
+  We want to ensure that all of the libraries show in
+  /var/lib/snapd/lib/gl{,32} can be loaded with dlopen without errors.
+
+  While not a perfect test, it shows the extent of the contract that snapd
+  promises to upkeep - that the libraries exist and may be loaded.
+systems:
+  - ubuntu-24.04-64
+environment:
+  PACKAGE_VERSION/535: 535
+  PACKAGE_VERSION/550: 550
+prepare: |
+  ./task.sh prepare
```

Comment:
Why split the test to different file?

Reply:
Syntax highlighting, `shfmt`, trivial shellcheck integration.

Reply:
still not convinced of having the test in a different file, mostly for consistency. Perhaps we could create a tool to apply shfmt to the prepare, execute and restore parts in the task.yaml (as we did for shellcheck)?

Reply:
Doing something that we've been doing for a long time is consistent but doesn't mean it's a good thing. I gave three specific reasons why this is better. I can only say that from the point of view of developing the test it's superior than what we started with. The more code we have the more shoving it all in yaml is more fragile and requires us to invest in custom tools nobody gets right (shellcheck validation script is a huge broken hack).